lib/if.c

```diff
@@ -1100,7 +1100,10 @@ DEFPY_NOSH (interface,
 	 * VRF.
 	 */
 	VRF_GET_ID(vrf_id, vrfname, false);
-	ifp = if_lookup_by_name_all_vrf(ifname);
+	if (vrf_get_backend() == VRF_BACKEND_VRF_LITE)
+		ifp = if_lookup_by_name_all_vrf(ifname);
+	else
+		ifp = if_lookup_by_name(ifname, vrf_id);
```

Comment:
Wouldn't it just be best to use `ifp = if_lookup_by_name(ifname, vrf_id);`   Why do we need *two* different ways of doing it

Reply:
why ? backward compatibility for people that don't enter the vrf keywork when reaching an interface.

Reply:
This is not about the user, it's about the internal programming api.  If in order to lookup an interface I have to sprinkle if else statements throughout the code base then the interface is *wrong* and needs to be fixed.  I'm asking that an actual interface be one call and it do the right thing.  If we cannot do that we have a real problem

Reply:
one more time, behaviour is at startup.
at startup, you may have vrf_get_backend is unknown, then the vty interface used for executing configuration must rely on a model accurate.

Reply:
As much as I don't like it (added complexity), I think this change is necessary.

Using `if_lookup_by_name_all_vrf()` with the netns VRF backend can lead to problems when interface names aren't globally unique among netnses, so `if_lookup_by_name()` needs to be used instead.

For the VRF-lite backend, however, we want to use `if_lookup_by_name_all_vrf()` to preserve the behavior of finding the interface VRF automatically when one isn't specified (as Philippe explained).

I only think the new comment in the code is confusing and doesn't add much value, would prefer if it could be improved or removed altogether.

Reply:
removing the comment

Reply:
@pguibert6WIND please pull this commit to a separate PR as it's controversial and needs to be reviewed in more detail.